src/NuGet.Core/NuGet.Configuration/Settings/Settings.cs

```diff
@@ -715,7 +715,9 @@ private static string ResolveRelativePath(string originDirectoryPath, string ori
                 return path;
             }
 
-            return Path.Combine(originDirectoryPath, ResolvePath(Path.GetDirectoryName(originFilePath), path));
+            var rawPath = Path.Combine(originDirectoryPath, ResolvePath(Path.GetDirectoryName(originFilePath), path));
```

Comment:
Can you elaborate what scenario failed?

Reply:
Is originDirectoryPath not rooted?

Reply:
When `nuget.config` says that the package source is `.\nupkgs`, this method returns `c:\path\to\repo\.\nupkgs`, and V2 restore logic uses this value, which is then passed to package source telemetry. However, V3 restore logic normalizes the directory name, and creates new `PackageSource` objects, so when V3 sends package source telemetry, the source string does not match.  This changes makes sure it always matches.